crates/contracts/src/world/world_contract.cairo

```diff
@@ -819,11 +977,19 @@ pub mod world {
         ///
         /// * `keys` - The keys of the event.
         /// * `values` - The data to be logged by the event.
-        fn emit(self: @ContractState, mut keys: Array<felt252>, values: Span<felt252>) {
-            let system = get_caller_address();
-            system.serialize(ref keys);
-
-            emit_event_syscall(keys.span(), values).unwrap_syscall();
+        fn emit(
+            ref self: ContractState,
+            event_selector: felt252,
+            keys: Span<felt252>,
+            values: Span<felt252>,
+            historical: bool
```

Comment:
I'm wondering if the historical should come from the calldata or something we defined when registering the event. @Larkooo could it make sense for the user to toggle this?

Reply:
At the moment, the `historical` flag is provided through the `dojo::event` attribute. That means, it is readable from the event contract (with `historical()`). An option would be to call the event contract in the `emit()` function to get the value of the `historical` flag. So, no need to provide it when registering the event nor when calling the `emit()` function.

What do you think ?

Reply:
Since we will add permissions, passing the boolean is effectively the cheapest way to do it, instead of reading additional data from the storage.

So let's keep it like this for the first iteration I think.